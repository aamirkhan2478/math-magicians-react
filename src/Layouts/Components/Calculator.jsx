import React, { useState } from "react";
import calculate from "../../logic/calculate";
import "../Assets/styles/calculator.css";

const Calculator = () => {
  const [state, setState] = useState({
    total: 0,
    next: null,
    operation: null,
  });
  const { total, next, operation } = state;

  const updateCalc = (e) => {
    setState((state) => calculate(state, e.target.textContent));
  };

  return (
    <div className="container">
      <div className="text-box">
        {total}
        {operation}
        {next}
      </div>
      <div className="grid-container">
        <button className="btn" value="AC" onClick={updateCalc}>
          AC
        </button>
        <button className="btn" onClick={updateCalc}>
          +/-
        </button>
        <button className="btn" onClick={updateCalc}>
          %
        </button>
        <button className="btn btn-orange" onClick={updateCalc}>
          +
        </button>
        <button className="btn" onClick={updateCalc}>
          7
        </button>
        <button className="btn" onClick={updateCalc}>
          8
        </button>
        <button className="btn" onClick={updateCalc}>
          9
        </button>
        <button className="btn btn-orange" onClick={updateCalc}>
          x
        </button>
        <button className="btn" onClick={updateCalc}>
          4
        </button>
        <button className="btn" onClick={updateCalc}>
          5
        </button>
        <button className="btn" onClick={updateCalc}>
          6
        </button>
        <button className="btn btn-orange" onClick={updateCalc}>
          &divide;
        </button>
        <button className="btn" onClick={updateCalc}>
          1
        </button>
        <button className="btn" onClick={updateCalc}>
          2
        </button>
        <button className="btn" onClick={updateCalc}>
          3
        </button>
        <button className="btn btn-orange" onClick={updateCalc}>
          -
        </button>
        <button className="btn btn-block" onClick={updateCalc}>
          0
        </button>
        <button className="btn" onClick={updateCalc}>
          .
        </button>
        <button className="btn btn-orange" onClick={updateCalc}>
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
