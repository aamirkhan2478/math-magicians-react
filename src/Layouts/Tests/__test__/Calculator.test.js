import renderer from 'react-test-renderer';
import Calculator from '../../Components/Calculator';

it('renders calculator correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
