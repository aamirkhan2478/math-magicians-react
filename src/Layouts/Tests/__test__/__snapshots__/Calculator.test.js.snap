// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator correctly 1`] = `
<div
  className="container"
>
  <div
    className="text-box"
  >
    0
  </div>
  <div
    className="grid-container"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn btn-orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn btn-orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn btn-orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn btn-orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn btn-block"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn btn-orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
